tt_metal/Glossary.md

```diff
@@ -0,0 +1,145 @@
+- **Deterministic Programming -** Programming style where the same input will always produce the same output.
+- **dim -** Dimension.
+- **DPRINT -** Configurable code formatting platform.
+- **dtype -** Data Type.
+- **Elt-wise -** Element Wise - A programming technique that performs an operation on each element of a data container independently.
+- **Ethernet Core -** Facilitates chip-to-chip communication across the mesh.
+- **FeedForward -** Characteristic of a network. Feedforward networks provide context for data before sending the data. When expected data is received, confirmation feedback is provided.
+- **HelloWorld -** Basic computer program often used as a starting point.
+- **Huge Page -** Memory management technique used to request larger memory blocks.
+- **Kernel Types -** Kernels send and receive data from tensors.
+  - **Bare Metal C/C++ Kernels -** Kernels written in C or C++ for execution. Generally used to accelerate computation and processing performance.
+  - **Reader Kernel -** Kernel receiving data from a DRAM or SRAM buffer.
+  - **Writer Kernel -** Kernel sending data to a DRAM or SRAM buffer.
+  - **User Kernel Types -** Kernels are in user mode when operating in applications.
+    - **Compute Kernels -** Kernels used for processing tasks or operations. Compute kernels will automatically generate the following types of kernels:
+      - **Unpack Kernels -** Unpack kernels prepare data for operations to be performed by the math kernel.
+      - **Math Kernels -** Kernels used for matrix multiplication and other mathematical tasks or operations.
+      - **Pack Kernels -** Pack kernels wait for the end of the math kernel and prepare data to be moved to the next part of the system.
+    - **Data Movement Kernels _** The first and fifth RISC-Vs on a Tensix Core responsible for moving data between NoCs, memory buffers, and the compute kernel.
+    - **Ethernet Data Movement Kernels -** Kernels responsible for moving data between cores.
+  - **Dispatch Kernels -** Kernels that determine where data is dispatched depending on the data’s priority and dispatch key.
+  - **Low-Level Kernels -** Kernels at the bottom of the software stack, these are for the system’s basic functions.
+- **Mantissa -** Floating point number that represents significant digits of the number.
+- **Memory Types -** Memory can be Interleaved or Sharded and can be further categorized into height, width, and block sharding.
+  - **Interleaved Memory -** L1 interleaved distributes across all L1 memory banks sequentially.
+  - **Sharded Memory -** L1 sharded lays out tensors based user provided shard specifications.
+      - **Height Sharding -** L1 Memory split across dimension 0.
+      - **Width Sharding -** L1 Memory split across dimension 1.
+      - **Block Sharding -** L1 Memory split across dimension 0 and dimension 1.
```

Comment:
Sharded can be L1 or DRAM. I am not sure what the descriptions of height, width, block sharding mean at all. What does `L1 Memory split across dim` mean?